### ISSUE TITLE
[Expo52] Error: pod install exited with non-zero code: 31

### ISSUE THREAD

**Issue by boh001** [expo/expo#33057]
### Minimal reproducible example

https://github.com/boh001/expo-example

### What platform(s) does this occur on?

iOS

### Where did you reproduce the issue?

in a development build

### Summary

when run `eas build --local --platform ios --profile development`, i get this error. but `--platform android` is fine.

```bash
Error: pod install exited with non-zero code: 31
    at ChildProcess.completionListener (/Users/sanghyeon/.npm/_npx/203b5a3d3de594cf/node_modules/@expo/spawn-async/build/spawnAsync.js:42:23)
    at Object.onceWrapper (node:events:629:26)
    at ChildProcess.emit (node:events:514:28)
    at maybeClose (node:internal/child_process:1105:16)
    at Socket.<anonymous> (node:internal/child_process:457:11)
    at Socket.emit (node:events:514:28)
    at Pipe.<anonymous> (node:net:337:12)
    ...
    at spawnAsync (/Users/sanghyeon/.npm/_npx/203b5a3d3de594cf/node_modules/@expo/spawn-async/build/spawnAsync.js:7:23)
    at spawn (/Users/sanghyeon/.npm/_npx/203b5a3d3de594cf/node_modules/@expo/turtle-spawn/dist/index.js:16:47)
    at installPods (/Users/sanghyeon/.npm/_npx/203b5a3d3de594cf/node_modules/@expo/build-tools/dist/ios/pod.js:12:50)
    at runInstallPodsAsync (/Users/sanghyeon/.npm/_npx/203b5a3d3de594cf/node_modules/@expo/build-tools/dist/builders/ios.js:225:70)
    at /Users/sanghyeon/.npm/_npx/203b5a3d3de594cf/node_modules/@expo/build-tools/dist/builders/ios.js:77:19
    at BuildContext.runBuildPhase (/Users/sanghyeon/.npm/_npx/203b5a3d3de594cf/node_modules/@expo/build-tools/dist/context.js:107:34)
    at buildAsync (/Users/sanghyeon/.npm/_npx/203b5a3d3de594cf/node_modules/@expo/build-tools/dist/builders/ios.js:76:19)
    at async runBuilderWithHooksAsync (/Users/sanghyeon/.npm/_npx/203b5a3d3de594cf/node_modules/@expo/build-tools/dist/builders/common.js:12:13)
    at async Object.iosBuilder (/Users/sanghyeon/.npm/_npx/203b5a3d3de594cf/node_modules/@expo/build-tools/dist/builders/ios.js:35:16)
    at async buildIosAsync (/Users/sanghyeon/.npm/_npx/203b5a3d3de594cf/node_modules/eas-cli-local-build-plugin/dist/ios.js:44:12)

Build failed
Unknown error. See logs of the Install pods build phase for more information.
npx -y eas-cli-local-build-plugin@1.0.144 eyJqb2Ii...
```

### Environment

```text
expo-env-info 1.2.1 environment info:
    System:
      OS: macOS 14.5
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 20.10.0 - ~/Library/Caches/fnm_multishells/67486_1732010824005/bin/node
      Yarn: 1.22.22 - ~/Library/Caches/fnm_multishells/67486_1732010824005/bin/yarn
      npm: 10.2.3 - ~/Library/Caches/fnm_multishells/67486_1732010824005/bin/npm
    Managers:
      CocoaPods: 1.15.2 - /Users/sanghyeon/.rbenv/shims/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 24.1, iOS 18.1, macOS 15.1, tvOS 18.1, visionOS 2.1, watchOS 11.1
    IDEs:
      Android Studio: 2024.1 AI-241.18034.62.2412.12266719
      Xcode: 16.1/16B40 - /usr/bin/xcodebuild
    npmPackages:
      expo: ~52.0.7 => 52.0.7 
      expo-router: ~4.0.6 => 4.0.6 
      react: 18.3.1 => 18.3.1 
      react-dom: 18.3.1 => 18.3.1 
      react-native: 0.76.2 => 0.76.2 
      react-native-web: ~0.19.13 => 0.19.13 
    npmGlobalPackages:
      eas-cli: 13.3.0
    Expo Workflow: managed
```

### Expo Doctor Diagnostics

```text
✔ Check package.json for common issues
✔ Check Expo config for common issues
✔ Check native tooling versions
✔ Check if the project meets version requirements for submission to app stores
✔ Check for common project setup issues
✔ Check dependencies for packages that should not be installed directly
✔ Check for app config fields that may not be synced in a non-CNG project
✔ Check for issues with Metro config
✔ Validate packages against React Native Directory package metadata
✔ Check npm/ yarn versions
✔ Check Expo config (app.json/ app.config.js) schema
✔ Check that packages match versions required by installed Expo SDK
✔ Check for legacy global CLI installed locally
✔ Check that native modules do not use incompatible support packages
✔ Check that native modules use compatible support package versions for installed Expo SDK
```

**Comment by brentvatne** [comment]
what do you see if you run `npx expo prebuild -p ios`?

**Comment by boh001** [comment]
Thank you for your reply, but I resolved it! Updating the Ruby version(3.3.0-dev -> 3.3.5) fixed the issue.
